src/components/Reports/GDPRPermissions/SubscriberGdprPermissions.js

```diff
@@ -1,8 +1,20 @@
-import React, { useState, useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 import { InjectAppServices } from '../../../services/pure-di';
 import { Loading } from '../../Loading/Loading';
 import { FormattedMessage, useIntl } from 'react-intl';
 import PermissionExpandableRow from './PermissionExpandableRow';
+import { searchLinkByRel } from '../../../utils';
+import { DownloadLink } from './SubscriberGdpr.styles';
+
+const permissionHistoryLinkRel = 'docs/rels/get-permission-history-csv';
+
+const getDownloadUrl = (links) => {
+  if (links) {
+    const link = searchLinkByRel(links, permissionHistoryLinkRel)[0];
```

Comment:
We do not consider this logic as _presentation logic_, so, in other cases, we do not do this at the _component level_.

In place of this we parse the links at the _service level_, for example: 

https://github.com/FromDoppler/doppler-webapp/blob/eb1c3ea03a97b997f705e0fd8bf72ee326a260d8/src/services/doppler-billing-api-client.ts#L71-L78

https://github.com/FromDoppler/doppler-webapp/blob/eb1c3ea03a97b997f705e0fd8bf72ee326a260d8/src/services/doppler-billing-api-client.ts#L80-L93